runtime/bcverify/staticverify.c

```diff
@@ -1753,6 +1753,9 @@ j9bcv_verifyClassStructure (J9PortLibrary * portLib, J9CfrClassFile * classfile,
 				errorType = J9NLS_CFR_ERR_BAD_METHOD_NAME__ID;
 				goto _formatError;
 			}
+#if JAVA_SPEC_VERSION >= 18
+			isInit = isInit && (CFR_CONSTANT_InvokeDynamic != info->tag);
```

Comment:
change looks good, I have a nitpick with overwriting `isInit`. It makes it confusing from a clarity perspective.

I would prefer:
```
			if ((isInit)
#if JDK18  
                        && (CFR_CONSTANT_InvokeDynamic != info->tag)
#endif /* JDK18 */
                        ) {
				U_16 returnChar = getReturnTypeFromSignature(info->bytes, info->slot1, NULL);
				if ((info->bytes[info->slot1 - 1] != 'V') && !IS_QTYPE(returnChar)) {
					errorType = J9NLS_CFR_ERR_BC_METHOD_INVALID_SIG__ID;
					goto _formatError;
				}
			}
```

Reply:
I would leave the checks as is and add a new local

Reply:
Updated.